src/rotator_library/cost_efficiency.py

```
@@ -0,0 +1,300 @@
+
+        `priority`: "low" | "normal" | "high". Only archetype A uses it (low
+        priority -> bias toward cheap proxy; high -> quality).
+        """
+        cands = [m for m in models if m.get("quality", 0.0) >= quality_floor]
+        if not cands:
+            cands = list(models)  # quality_floor wiped everything -> ignore it
+        arche = self.classify_provider(provider_id)
+
+        if arche == "B":
+            # flat quota: model choice doesn't change call budget -> best quality.
+            key = lambda m: (-m.get("quality", 0.0), m.get("model_id", ""))
```

Ruff (0.15.18)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `key` as a `def`

(E731)